realsense2_camera/src/profile_manager.cpp

```diff
@@ -139,6 +139,43 @@ std::map<stream_index_pair, rs2::stream_profile> ProfilesManager::getDefaultProf
     return sip_default_profiles;
 }
 
+rs2::stream_profile VideoProfilesManager::validateAndGetSuitableProfile(rs2_stream stream_type, rs2::stream_profile given_profile)
+{
+    if (_all_profiles.empty()) 
+        throw std::runtime_error("Wrong commands sequence. No profiles set.");
+
+    rs2::stream_profile suitable_profile = given_profile;
+    bool is_given_profile_suitable = false;
+
+    for (auto temp_profile : _all_profiles)
+    {
+        if (temp_profile.stream_type() == stream_type)
+        {
+            auto video_profile = given_profile.as<rs2::video_stream_profile>();
+
+            if (isSameProfileValues(temp_profile, video_profile.width(), video_profile.height(), video_profile.fps(), RS2_FORMAT_ANY))
+            {
+                is_given_profile_suitable = true;
+                break;
+            }
+        }
+    }
+
+    // If given profile is not suitable, choose the first available profile for the given stream.
+    if (!is_given_profile_suitable)
```

Comment:
and also here, can you give me an example ?

Reply:
Depth Mapping Camera module doesn't have any default profiles (for both occupancy and labeled pointcloud streams). So, `getDefaultProfiles()` function will consider the first listed profile as default profile. Due to this reason, lpc stream's profile will be considered as default profile. But that profile is not supported by Occupancy stream. 

So here, for Occupancy stream, the given profile (lpc's profile) is not suitable. If not suitable, consider the first detected Occupancy stream profile.

Let's apply same scenario for Depth module. Depth stream has default profile, but IR streams doesn't. Fortunately, depth stream's default profile is suitable for IR streams. No issues for this scenario.

Reply:
This case can be handled either this way or inside `getDefaultProfiles()` function itself by providing one default profile per stream.